sqlalchemy_dremio/base.py

```diff
@@ -0,0 +1,266 @@
+
+class AcInteger(types.Integer):
+    def get_col_spec(self):
+        return "INTEGER"
+
+class AcTinyInteger(types.Integer):
+    def get_col_spec(self):
+        return "TINYINT"
+
+class AcSmallInteger(types.SmallInteger):
+    def get_col_spec(self):
+        return "SMALLINT"
+
+class AcDateTime(types.DateTime):
+    def get_col_spec(self):
+        return "DATETIME"
+
+class AcDate(types.Date):
+
+    def get_col_spec(self):
+        return "DATETIME"
+
+class AcText(types.Text):
+    def get_col_spec(self):
+        return "MEMO"
+
+class AcString(types.String):
+    def get_col_spec(self):
+        return "TEXT" + (self.length and ("(%d)" % self.length) or "")
+
+class AcUnicode(types.Unicode):
+    def get_col_spec(self):
+        return "TEXT" + (self.length and ("(%d)" % self.length) or "")
+
+    def bind_processor(self, dialect):
+        return None
+
+    def result_processor(self, dialect, coltype):
+        return None
+
+class AcChar(types.CHAR):
+    def get_col_spec(self):
+        return "TEXT" + (self.length and ("(%d)" % self.length) or "")
+
+class AcBinary(types.LargeBinary):
+    def get_col_spec(self):
+        return "BINARY"
+
+class AcBoolean(types.Boolean):
+    def get_col_spec(self):
+        return "YESNO"
+
+class AcTimeStamp(types.TIMESTAMP):
+    def get_col_spec(self):
+        return "TIMESTAMP"
+
+class DremioExecutionContext(default.DefaultExecutionContext):
+
+    def get_lastrowid(self):
+        self.cursor.execute("SELECT @@identity AS lastrowid")
+        return self.cursor.fetchone()[0]
+
+
+class DremioCompiler(compiler.SQLCompiler):
+    extract_map = compiler.SQLCompiler.extract_map.copy()
+    extract_map.update({
+            'month': 'm',
+            'day': 'd',
+            'year': 'yyyy',
+            'second': 's',
+            'hour': 'h',
+            'doy': 'y',
+            'minute': 'n',
+            'quarter': 'q',
+            'dow': 'w',
+            'week': 'ww'
+    })
+
+    def visit_cast(self, cast, **kwargs):
+        return cast.clause._compiler_dispatch(self, **kwargs)
+
+    def visit_select_precolumns(self, select):
+        """Dremio puts TOP, it's version of LIMIT here """
+        s = select.distinct and "DISTINCT " or ""
+        if select.limit:
+            s += "TOP %s " % (select.limit)
+        if select.offset:
+            raise exc.InvalidRequestError(
+                    'Dremio does not support LIMIT with an offset')
+        return s
+
+    def limit_clause(self, select):
+        """Limit in dremio is after the select keyword"""
+        return ""
+
+    def binary_operator_string(self, binary):
+        """Dremio uses "mod" instead of "%" """
+        return binary.operator == '%' and 'mod' or binary.operator
+
+    function_rewrites = {'current_date': 'now',
+                          'current_timestamp': 'now',
+                          'length': 'len',
+                          }
+    def visit_function(self, func, **kwargs):
```

Comment:
Dremio function names do not often differ from ANSI SQL names. 
Very misleading comments tied to code that should be ejected throughout this section that should be excised given how we DON'T want to override typical SQL for Dremio.